Applications/Spire/Source/KeyBindings/CancelKeyBindingsModel.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Spire/KeyBindings/CancelKeyBindingsModel.hpp"
+#include "Spire/Spire/ValidatedValueModel.hpp"
+
+using namespace boost;
+using namespace Spire;
+
+CancelKeyBindingsModel::CancelKeyBindingsModel() {
+  for(auto i = 0; i < 13; ++i) {
+    auto operation = static_cast<Operation>(i);
+    m_bindings[operation] = make_validated_value_model<QKeySequence>(
+      std::bind_front(&CancelKeyBindingsModel::on_validate, this, operation),
+        std::make_shared<LocalKeySequenceValueModel>());
+  }
+}
+
+std::shared_ptr<KeySequenceValueModel>
+    CancelKeyBindingsModel::get_binding(Operation operation) const {
+  return m_bindings.at(operation);
+}
+
+optional<CancelKeyBindingsModel::Operation>
+    CancelKeyBindingsModel::find_operation(const QKeySequence& sequence) const {
+  for(auto& binding : m_bindings) {
+    if(binding.second->get() == sequence) {
+      return binding.first;
+    }
+  }
+  return {};
+}
+
+QValidator::State CancelKeyBindingsModel::on_validate(Operation operation,
+    const QKeySequence& sequence) {
+  if(sequence.count() == 0) {
+    return QValidator::Intermediate;
+  } else if(sequence.count() > 1) {
+    return QValidator::Invalid;
+  }
+  if(auto search = find_operation(sequence); search && *search != operation) {
```

Comment:
Can you explain this? The API states that in a scenario where an operation is bound to an existing key binding, the existing key binding is reset, but it looks like this will result in a rejection. The correct behavior is to reset the existing keybinding.